src/osp/Active/SysPhysics.h

```diff
@@ -35,22 +36,32 @@ class SysPhysics
 {
 public:
 
-    enum EIncludeRootMass { Ignore, Include };
+    static void calculate_subtree_mass_center(
+            acomp_storage_t<ACompTransform> const&  rTf,
+            ACtxPhysics&                            rCtxPhys,
+            ACtxSceneGraph&                         rScnGraph,
+            ActiveEnt                               root,
+            Vector3&                                rMassPos,
+            float&                                  rTotalMass,
+            Matrix4 const&                          currentTf = {});
 
-    // TODO: rewrite hierarchy inertia calculations for new mass/inertia system
+    static void calculate_subtree_mass_inertia(
+            acomp_storage_t<ACompTransform> const&  rTf,
+            ACtxPhysics&                            rCtxPhys,
+            ACtxSceneGraph&                         rScnGraph,
+            ActiveEnt                               root,
+            Matrix3&                                rInertiaTensor,
+            Matrix4 const&                          currentTf = {});
 
     template<typename IT_T>
-    static void update_delete_phys(
-            ACtxPhysics &rCtxPhys, IT_T first, IT_T const& last);
+    static void update_delete_phys(ACtxPhysics& rCtxPhys, IT_T const& first, IT_T const& last);
```

Comment:
consider changing the template type for the iterators to have a separate type for the first vs the last, to allow for the C++20 "sentinel" iterator feature to work.